#!/usr/bin/env python
PACKAGE = "zm_robot_control_pid"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("wheel1_P", double_t, 0, "leaft after wheel PID Controller P parameter", 10.0, 0, 1000)
gen.add("wheel1_I", double_t, 0, "leaft after wheel PID Controller I parameter",    0, 0, 1000)
gen.add("wheel1_D", double_t, 0, "leaft after wheel PID Controller D parameter",    0, 0, 1000)

gen.add("wheel2_P", double_t, 0, "leaft front wheel PID Controller P parameter", 10.0, 0, 1000)
gen.add("wheel2_I", double_t, 0, "leaft front wheel PID Controller I parameter",    0, 0, 1000)
gen.add("wheel2_D", double_t, 0, "leaft front wheel PID Controller D parameter",    0, 0, 1000)

gen.add("wheel3_P", double_t, 0, "right front wheel PID Controller P parameter", 10.0, 0, 1000)
gen.add("wheel3_I", double_t, 0, "right front wheel PID Controller I parameter",    0, 0, 1000)
gen.add("wheel3_D", double_t, 0, "right front wheel PID Controller D parameter",    0, 0, 1000)

gen.add("wheel4_P", double_t, 0, "right after wheel PID Controller P parameter", 10.0, 0, 1000)
gen.add("wheel4_I", double_t, 0, "right after wheel PID Controller I parameter",    0, 0, 1000)
gen.add("wheel4_D", double_t, 0, "right after wheel PID Controller D parameter",    0, 0, 1000)

exit(gen.generate(PACKAGE, "zm_robot_control_pid", "ZMPIDController"))
